server/market/market.go

```diff
@@ -1051,15 +1051,20 @@ func (m *Market) collectPreimages(orders []order.Order) (cSum []byte, ordersReve
 		}
 
 		// Failure to respond in time is a miss, signalled by a nil pointer.
+		var missOnce sync.Once
 		miss := func() {
-			piChan <- nil
+			// RequestWithTimeout should only call this on expire, not because
+			// of link or other errors, but put it in a sync.Once to be safe
+			// since it will block if run twice.
+			missOnce.Do(func() { piChan <- nil })
 		}
 
 		// Send the preimage request to the order's owner.
 		err = m.auth.RequestWithTimeout(ord.User(), req, func(_ comms.Link, msg *msgjson.Message) {
 			m.handlePreimageResp(msg, reqData)
 		}, piTimeout, miss)
 		if err != nil {
+			miss() // only called by RequestWithTimeout on expire, not send errors
```

Comment:
Was a bug where a request error (not just expire) would prevent a result for this requests, preventing epoch processing.